content/en/docs/architecture/ci-operator-internals/testing.md

```diff
@@ -0,0 +1,425 @@
+brief discussion that follows:
+
+- [ProwJobs][prowjobs]
+- [Life of a Prow Job][life_of_a_prow_job]
+
+For our purposes, it is enough to understand that `ci-operator` needs to
+determine what source code will be used as the primary input for its execution.
+This ultimately takes the form of a `git` repository and one or more
+<i>ref</i>s/revisions (see [`gitglossary(7)`][gitglossary] and
+[`gitrevisions(7)`][gitrevisions] for the definition of these terms).  In the
+context of a CI job, this information is supplied by Prow via the `JOB_SPEC`
+variable (as described in [this section][job_env_variables] of the document
+linked above).
+
+Two options are available to execute `ci-operator` directly, outside of Prow.
+The `--git-ref` argument can be used to pass a string in the format
+`org/repo@ref`, which instructs `ci-operator` to `ls-remote` the repository to
+determine the revisions.  It then build its own `JOB_SPEC` internally based on
+that information:
+
+{{< highlight bash >}}
+$ ci-operator --git-ref openshift/ci-tools@master |& head -n 2
+INFO[2022-03-09T17:01:38Z] unset version 0
+INFO[2022-03-09T17:01:39Z] Resolved openshift/ci-tools@master to commit ef415037c6f8942784ee8a15635d5f0ce8a9cfcc
+{{< / highlight >}}
+
+The second option is to manufacture a `JOB_SPEC` conforming to the format
+dictated by Prow to fit the desired test scenario.  This saves some time when
+testing several executions and can also be useful when testing the processing of
+this variable or alternative configurations.  `JOB_SPEC` is a JSON object which
+can be easily written manually.  For now, even a minimal definition suffices.
+However, code in Prow and `ci-tools` generally assumes a well-formed definition,
+so it is best to supply a valid value unless something very specific is begin
+tested.  The following examples assume a definition similar to what the
+`--git-ref` argument shown above generates:
+
+{{< highlight bash >}}
+export JOB_SPEC='{
+  "type": "postsubmit",
+  "refs": {
+    "org": "openshift",
+    "repo": "ci-tools",
+    "base_ref": "master",
+    "base_sha": "ef415037c6f8942784ee8a15635d5f0ce8a9cfcc"
+  },
+  "decoration_config": {
+    "gcs_configuration": {
+      "bucket": "origin-ci-test",
+      "path_strategy": "single",
+      "default_org": "openshift",
+      "default_repo": "origin",
+      "mediaTypes": {"log": "text/plain"}
+    },
+    "utility_images": {
+      "clonerefs": "registry.ci.openshift.org/ci/clonerefs:latest",
+      "initupload": "registry.ci.openshift.org/ci/initupload:latest",
+      "entrypoint": "registry.ci.openshift.org/ci/entrypoint:latest",
+      "sidecar": "registry.ci.openshift.org/ci/sidecar:latest"
+    }
+  }
+}'
+{{< / highlight >}}
+
+## `ci-operator-configresolver`
+
+In a pattern that will soon become recurrent, attempting to execute
+`ci-operator` with this new definition still results in an error, but a
+different one:
+
+{{< highlight bash >}}
+$ ci-operator
```

Comment:
No, this is the correct invocation here.  It is meant to show what happens when no arguments are passed.

The document is structured so that it shows the situations someone starting to test `ci-operator` may encounter and explains why they happen.